ethereum/test/contractBackend.go

```diff
@@ -0,0 +1,63 @@
+package test
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	ethchannel "perun.network/go-perun/backend/ethereum/channel"
+	ethchanneltest "perun.network/go-perun/backend/ethereum/channel/test"
+	ethwallet "perun.network/go-perun/backend/ethereum/wallet"
+	"perun.network/go-perun/wallet"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/stretchr/testify/require"
+
+	"github.com/direct-state-transfer/dst-go/ethereum"
+	"github.com/direct-state-transfer/dst-go/identity"
+)
+
+//DefaultTxTimeout is the default transaction timeout for simulated backend.
+const DefaultTxTimeout = 5 * time.Second
+
+type ContractBackendSetup struct {
+	WalletBackendSetup
+	ContractBackend    *ethereum.ContractBackend
+	AdjAddr, AssetAddr wallet.Address
+}
+
+// NewContractBackendSetup initializes and returns a test ContractBackend.
+// It returns an initialized user with test accounts and a simulated contract backend,
+// with adjudicator and asset holder contracts deployed on it.
+func NewContractBackendSetup(t *testing.T, rng *rand.Rand, tempDir string) (ContractBackendSetup, func()) {
```

Comment:
Is it important that the `tempDir` can be set or what is wrong with having it random?

Reply:
`tempDir` is the temporary directory in which temp data for testing will be created. In this case, it will be the `testdata` directory in the project root. 
I have updated the tests to use system's temp directory and hence this argument has been removed.